src/tasks/includes/lint-reporter.js

```diff
@@ -0,0 +1,52 @@
+const gutil = require('gulp-util');
+
+/** Class representing a custom reporter for @shopify/theme-lint */
+export default class Reporter {
+  constructor() {
+    this.successes = [];
+    this.failures = [];
+  }
+
+  /**
+   * Pushes a valid message onto successes.
+   *
+   * @param {String} message
+   * @param {String} file
+   */
+  success(message, file = null) {
```

Comment:
Both `success` and `failure` take an third optional `index` argument to indicate the error's location in the file.